metrics/local_test.go

```diff
@@ -10,11 +9,6 @@ import (
 )
 
 func TestLocalMetrics(t *testing.T) {
-	numGoroutines := runtime.NumGoroutine()
-	defer func() {
-		assert.Equal(t, numGoroutines, runtime.NumGoroutine(), "Leaked at least one goroutine.")
-	}()
```

Comment:
Do you think there's value in doing this? adding an atomic var/waitgroup just to ensure that the function exited?